nirps/qc_checks/check_astrometric_database.py

```diff
@@ -0,0 +1,56 @@
+# Pierrot, 2024-03-12
+
+import pandas as pd
+import numpy as np 
+import requests
+from io import StringIO
+from astropy.table import Table
+
+def fetch_astrometric_database(main_url, pending_url):
+    # Fetch main table
+    main_request = requests.get(main_url)
+    main_dataframe = pd.read_csv(StringIO(main_request.text))
+
+    # Fetch pending table
+    pending_request = requests.get(pending_url)
+    pending_dataframe = pd.read_csv(StringIO(pending_request.text))
+
+    # Merge main and pending tables, removing duplicates
+    astrom_dataframe = pd.concat([main_dataframe, pending_dataframe]).drop_duplicates()
+
+    # Convert to Astropy table
+    astrom_table = Table.from_pandas(astrom_dataframe)
+
+    return astrom_table
+
+def check_objects_in_database(csv_file, database):
+    df = pd.read_csv(csv_file)
+    objects_not_in_database = [] # List of objects not in the database
+
+    # Check if objects are in the database
+    for obj_name in df['Object Name']:
+        found = False
+        for record in database:
+            if obj_name.lower() == record['OBJNAME'].lower() or obj_name.lower() in [alias.lower() for alias in record['ALIASES']]:
+                found = True
+                break
+        if not found:
+            objects_not_in_database.append(obj_name)
+
+    return objects_not_in_database
+
+MAIN_URL = ('https://docs.google.com/spreadsheets/d/'
```

Comment:
It's best practise to put global variables (that require the user to change) at the top of the code. This is better than having to search the code for them.

1. Description of code
2. Imports
3. Global variables

i.e.
```python
"""
Description of code
"""

import numpy as np


MAIN_URL = 'www.google.com'


def my_function(variable1: str, variable2: int) --> bool:
     """
     my function description

     :param variable1: str, variable1 description
     :param variable2: int, variable2 description

     :return: bool, return variable description
    """
    # my function code
    print(variable1)


if __name__ == '__main__':
     # my main code
     print('Hello World')

```